packages/wallet-ts/src/utils/wallets/welldone/WelldoneWallet.ts

```diff
@@ -0,0 +1,219 @@
+/* eslint-disable class-methods-use-this */
+import {
+  ChainId,
+  CosmosChainId,
+  TestnetCosmosChainId,
+} from '@injectivelabs/ts-types'
+import {
+  TxResponse,
+  TxRestApi,
+  CosmosTxV1Beta1Tx,
+  getPublicKey,
+} from '@injectivelabs/sdk-ts'
+import { DirectSignResponse, makeSignBytes } from '@cosmjs/proto-signing'
+import {
+  ErrorType,
+  UnspecifiedErrorCode,
+  CosmosWalletException,
+  WalletErrorActionModule,
+} from '@injectivelabs/exceptions'
+import { getEndpointsFromChainId } from '../cosmos/endpoints'
+import { WalletProvider } from './welldone'
+
+const $window = (typeof window !== 'undefined' ? window : {}) as Window & {
+  dapp?: WalletProvider
+}
+
+export class WelldoneWallet {
+  private chainId: CosmosChainId | TestnetCosmosChainId | ChainId
+
+  private chainName: string
+
+  constructor(chainId: CosmosChainId | TestnetCosmosChainId | ChainId) {
+    this.chainId = chainId
+    this.chainName = chainId.split('-')[0]
+  }
+
+  async getAccounts(): Promise<{ address: string; pubKey: string }> {
+    const welldone = await this.getWelldoneWallet()
+
+    try {
+      const accounts = await welldone.request(this.chainName, {
+        method: 'dapp:accounts',
+      })
+
+      if (Object.keys(accounts).length === 0) {
+        throw new Error(
+          `Please make the ${this.chainName} account in the WELLDONE wallet`,
+        )
+      }
+
+      return accounts[this.chainName]
+    } catch (e: unknown) {
+      throw new CosmosWalletException(new Error((e as any).message), {
+        contextModule: WalletErrorActionModule.GetAccounts,
+      })
+    }
+  }
+
+  public async broadcastTx(txRaw: CosmosTxV1Beta1Tx.TxRaw): Promise<string> {
+    const welldone = await this.getWelldoneWallet()
+
+    try {
+      const { hash } = await welldone.request(this.chainName, {
+        method: 'dapp:sendSignedTransaction',
+        params: [
+          `0x${Buffer.from(
+            CosmosTxV1Beta1Tx.TxRaw.encode(txRaw).finish(),
+          ).toString('hex')}`,
+        ],
+      })
+
+      return hash
+    } catch (e) {
+      throw new CosmosWalletException(new Error((e as any).message), {
+        context: 'WELLDONE',
+        contextModule: 'bradcast-tx',
+      })
+    }
+  }
+
+  public async waitTxBroadcasted(txHash: string): Promise<TxResponse> {
+    const endpoints = await this.getChainEndpoints()
+
+    return new TxRestApi(endpoints.rest).fetchTxPoll(txHash)
+  }
+
+  private changePubKeyType(authInfoBytes: Uint8Array) {
+    const authInfo = CosmosTxV1Beta1Tx.AuthInfo.decode(authInfoBytes)
+
+    authInfo.signerInfos.forEach((signerInfo) => {
+      if (signerInfo.publicKey) {
+        signerInfo.publicKey.typeUrl = '/cosmos.crypto.secp256k1.PubKey'
```

Comment:
@bangjelkoski 
WELLDONE Wallet use ‘secp245k1’ instead of ‘ethsecp256k1’ for unity with other cosmos chains. Therefore the injective account provided in the wallet is also made of ‘secp255k1’.

However, in the case of using `MsgBroadCaster`, the method `createSignerinfo()` is used in the process of making txRaw, and inside that method, it uses `getPublicKey()` that put publicKey’s type to '/injective.crypto.v1beta1.ethsecp256k1.PubKey' when the chainId starts with injective.
https://github.com/InjectiveLabs/injective-ts/blob/master/packages/sdk-ts/src/core/modules/tx/utils/tx.ts#L14

Therefore, in order to prepare for these exceptions, we added a part that changes the pubkey of authinfo to ‘secp256k1’ if you want to sign using WELLDONE Wallet. This is because we can use ‘stargate’ with the same environment as other cosmos chains.